### PR TITLE
[release-4.12] OCPBUGS-5749: Add namespace parameter and RBAC to WICD

### PR DESCRIPTION
**Pull request by saifshaikh48**
Manual backport of https://github.com/openshift/windows-machine-config-operator/pull/1369 (addressed a merge conflict in a unit test because https://github.com/openshift/windows-machine-config-operator/pull/1356 was not cleanly backported).

Fixes OCPBUGS-5749

### CODE REVIEW

File: pkg/daemon/controller/controller_test.go [hunk 1]
```diff
@@ -129,6 +131,93 @@ func TestResolveNodeVariables(t *testing.T) {
 	}
 }
 
+func TestResolvePowershellVariables(t *testing.T) {
```
Comment (saifshaikh48) [inline pkg/daemon/controller/controller_test.go:134]: The introduction of this unit test function was the only merge conflict. Everything else was cleanly cherry-picked.

**Comment by saifshaikh48** [comment]
/hold 

until [OCPBUGS-5065](https://issues.redhat.com//browse/OCPBUGS-5065) is verified

File: pkg/daemon/controller/controller_test.go (hunk 1 above)
Comment (sebsoto) [reply pkg/daemon/controller/controller_test.go:134]: Why is the unit test being introduced in this PR? It has nothing to do with it

Comment (sebsoto) [reply pkg/daemon/controller/controller_test.go:134]: Seems the correct thing to do here is drop this function completely

Comment (saifshaikh48) [reply pkg/daemon/controller/controller_test.go:134]: Keeping it prevents future merge conflicts related to this func. But I see your point -- the PR which first introduced function was not backported so why include it at all. I'll adjust the commit

**Comment by mtnbikenc** [comment]
[OCPBUGS-5065](https://issues.redhat.com//browse/OCPBUGS-5065) is verified
/hold cancel

**Comment by aravindhp** [comment]
/retest-required

vsphere-e2e-operator  [failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1375/pull-ci-openshift-windows-machine-config-operator-release-4.12-vsphere-e2e-operator/1613600737290358784) and there was an error in must-gather, so can't see logs to figure out what happened.

```
 2023/01/12 21:15:51 35m0.201881158s time is required to configure 1 nodes
    create_test.go:203: 
        	Error Trace:	/go/src/github.com/openshift/windows-machine-config-operator/test/e2e/create_test.go:203
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestWMCO/create/Creation/ConfigMap_controller/VM_is_configured_by_ConfigMap_controller
        	Messages:   	Windows node creation failed 
```

**Pull request merged**